[mlir][ArmSME] Add custom vector.print lowering for SME tiles

MacDue opened the pull request:
This adds a custom lowering for SME that loops over each row of the tile, extracting it via an SME MOVA, then printing with a normal 1D vector.print.

This makes writing SME integration tests easier and less verbose.

Depends on: #66910, #66911

MacDue commented:
cc @c-rhodes, @banach-space

c-rhodes commented on mlir/test/Integration/Dialect/Vector/CPU/ArmSME/tile_fill.mlir:
```diff
@@ -32,29 +32,15 @@ func.func @printTileEnd() {
-  vector.store %tile, %mem[%c0] : memref<?xi32>, vector<[4]x[4]xi32>
-
-  // Dump "mem". The smallest SVL is 128-bits so the tile will be at least
+  func.call @printTileBegin() : () -> ()
+  vector.print %tile : vector<[4]x[4]xi32>
+  func.call @printTileEnd() : () -> ()
```
move this below the CHECK lines

c-rhodes commented on mlir/lib/Dialect/ArmSME/Transforms/EnableArmStreaming.cpp:
```diff
@@ -61,7 +63,9 @@ struct EnableArmStreamingPass
   }
   void runOnOperation() override {
-    std::string attr;
+    if (getOperation()->getAttr(kArmEnableStreamingIgnore))
+      return;
```
Please could you add a test for this in `mlir/test/Dialect/ArmSME/enable-arm-streaming.mlir`

MacDue commented:
> Are you planning to break this up into separate PRs?

I plan on splitting adding the intrinsic and the `enable_arm_streaming` attribute into two separate PRs (with the tests you suggest). Once those are merged, I'll rebase and undraft this PR.

MacDue committed:
[mlir][ArmSME] Add custom vector.print lowering for SME tiles

This adds a custom lowering for SME that loops over each row of the
tile, extracting it via an SME MOVA, then printing with a normal 1D
vector.print.

This makes writing SME integration tests easier and less verbose.

dcaballe commented on mlir/lib/Conversion/ArmSMEToSCF/ArmSMEToSCF.cpp:
```diff
@@ -190,11 +190,94 @@ struct TileStoreOpConversion : public OpRewritePattern<arm_sme::TileStoreOp> {
+      // Loop body.
+      rewriter.setInsertionPointToStart(forOp.getBody());
+      // Extract the current row from the tile.
+      auto rowIndex = forOp.getInductionVar();
```
nit: please spell out the type if it's present on the right-hand side

dcaballe reviewed (approved):
Awesome, thanks!

c-rhodes reviewed (approved):
LGTM cheers

MacDue committed:
Review fixup

MacDue committed:
Update recent tests

c-rhodes commented:
is this ready to land?

MacDue commented:
> is this ready to land?

I think so :)
